integration_tests/src/main/python/regexp_test.py

```diff
@@ -243,6 +243,20 @@ def test_re_replace():
                 'REGEXP_REPLACE(a, "TEST", NULL)'),
         conf=_regexp_conf)
 
+def test_re_replace_choice_opt():
+    gen = mk_str_gen('[a-d]{1,2}') \
+        .with_special_case("cabd")
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark: unary_op_df(spark, gen).selectExpr(
+                'REGEXP_REPLACE(a, "ab|bc", "")',
+                'REGEXP_REPLACE(a, "ab|cd", "")',
+                'REGEXP_REPLACE(a, "ab|cd", "bc")',
+                'REGEXP_REPLACE(a, "ab|bc", "x")',
+                'REGEXP_REPLACE(a, "ab|cd", "b")',
+                'REGEXP_REPLACE(a, "ab|cd", "c")',
+                'REGEXP_REPLACE(a, "ab|cd", "x")'),
```

Comment:
I have no idea how can we verify if these expressions are actually executed by the code in this PR? I.e., how can verify if `ab|bc` will be called (on the GPU) by `regex_replace` or `replace(replace())`?